devdocs/protocols/tcp/kafka.md

```diff
@@ -0,0 +1,57 @@
+# OBI Kafka protocol parser
+
+This document describes the Kafka protocol parser that OBI provides.
+
+## Protocol Overview
+
+The Kafka protocol definition [Kafka protocol definition](https://Kafka.apache.org/protocol#protocol_messages) defines the schema and types of Kafka messages.
+Each message in Kafka starts with a header in the following format:
+
+```
+Request Header:
+  request_api_key => INT16
+  request_api_version => INT16
+  correlation_id => INT32
+  client_id => NULLABLE_STRING
+```
+
+the `request_api_key` defines the type of request, for example, `produce` or `fetch`.
+the `request_api_version` defines the version of the request, each message type has its own set of versions, and they increment independently.
+the `correlation_id` is used to correlate requests and responses.
+
+current request types OBI is tracking are:
+
+- *Produce (api key 0)*: OBI tracks these requests and produces `produce` spans.
+- *Fetch (api key 1)*: OBI tracks these requests and produces `consume` spans.
+- *Metadata (api key 3)*: OBI tracks these requests (and mainly the responses) to correlate topic names with fetch requests from v13 and above.
+  
+(unfortunately there is no way to anchor the documentation to a specific message.)
```

Comment:
Is this comment a leftover?

Reply:
maybe i can remove it, i just coudn't link the method stucture as a url since i can't anchor the name of the method, but i can remove it

Reply:
I think it's ok either way, was just making sure